server/src/main/java/org/opensearch/tasks/TaskResourceStats.java

```diff
@@ -32,27 +35,42 @@
  */
 public class TaskResourceStats implements Writeable, ToXContentFragment {
     private final Map<String, TaskResourceUsage> resourceUsage;
+    private final TaskThreadUsage threadUsage;
 
-    public TaskResourceStats(Map<String, TaskResourceUsage> resourceUsage) {
+    public TaskResourceStats(Map<String, TaskResourceUsage> resourceUsage, TaskThreadUsage threadUsage) {
         this.resourceUsage = Objects.requireNonNull(resourceUsage, "resource usage is required");
+        this.threadUsage = Objects.requireNonNull(threadUsage, "thread usage is required");
     }
 
     /**
      * Read from a stream.
      */
     public TaskResourceStats(StreamInput in) throws IOException {
         resourceUsage = in.readMap(StreamInput::readString, TaskResourceUsage::readFromStream);
+        if (in.getVersion().onOrAfter(Version.V_2_9_0)) {
+            threadUsage = TaskThreadUsage.readFromStream(in);
+        } else {
+            // Initialize TaskThreadUsage in case it is not found in mixed cluster case
+            threadUsage = new TaskThreadUsage(0, 0);
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeMap(resourceUsage, StreamOutput::writeString, (stream, stats) -> stats.writeTo(stream));
+        if (out.getVersion().onOrAfter(Version.V_2_9_0)) {
```

Comment:
This would need a fast follow up on `main` if I am not wrong.

Reply:
Yes! Already there https://github.com/opensearch-project/OpenSearch/pull/7889